packages/headless/src/features/product-listing/v2/product-listing-v2-actions-loader.ts

```diff
@@ -0,0 +1,133 @@
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+import {AsyncThunkProductListingV2Options} from '../../../api/commerce/product-listings/v2/product-listing-v2-api-client';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {productListingV2Reducer as productListing} from '../../../features/product-listing/v2/product-listing-v2-slice';
+import {ProductListingAction} from '../../analytics/analytics-utils';
+import {
+  logFacetClearAll,
+  logFacetDeselect,
+  logFacetSelect,
+  logFacetShowLess,
+  logFacetShowMore,
+  logFacetUpdateSort,
+  LogFacetDeselectActionCreatorPayload,
+  LogFacetSelectActionCreatorPayload,
+  LogFacetUpdateSortActionCreatorPayload,
+} from '../../facets/facet-set/facet-set-product-listing-analytics-actions';
+import {
+  FetchProductListingV2ThunkReturn,
+  fetchProductListing,
+  StateNeededByFetchProductListingV2,
+} from './product-listing-v2-actions';
+
+/**
+ * The product listing action creators.
+ */
+export interface ProductListingActionCreators {
+  /**
+   * Refreshes the product listing.
+   *
+   * @returns A dispatchable action.
+   */
+  fetchProductListing(): AsyncThunkAction<
+    FetchProductListingV2ThunkReturn,
+    void,
+    AsyncThunkProductListingV2Options<StateNeededByFetchProductListingV2>
+  >;
+}
+
+/**
+ * Loads the product listing reducer and returns the possible action creators.
+ *
+ * @param engine - The headless engine.
+ * @returns An object holding the action creators.
+ */
+export function loadProductListingActions(
+  engine: CommerceEngine
+): ProductListingActionCreators {
+  engine.addReducers({productListing});
+
+  return {
+    fetchProductListing,
+  };
+}
+
+/**
+ * The product listing analytics action creators.
+ */
+export interface ProductListingAnalyticsActionCreators {
```

Comment:
I think the methods defined on this interface should return `ProductListingV2Action`s instead of `ProductListingAction`s

Reply:
Agreed; this will require me to add some more files to the PR, though

(facet-set-product-listing-v2-analytics-actions.ts and probably facet-set-product-listing-utils.ts too).